lib/lcset.gi

```diff
@@ -0,0 +1,139 @@
+#A  AsociatedLeftCoset( <lc> ) . . . . . . forms a left coset of U by g \in G 
+##
+InstallMethod( AssociatedRightCoset, "generic method for left cosets", true, 
+    [ IsLeftCoset ], 0,
+function( lc )
+    local g, U, rc; 
+    g := Representative( lc ); 
+    U := ActingDomain( lc ); 
+    rc := RightCoset( U, g^-1 ); 
+    SetAssociatedLeftCoset( rc, lc ); 
+    return rc;
+end); 
+
+InstallMethod( AssociatedLeftCoset, "generic method for right cosets", true, 
+    [ IsRightCoset ], 0,
+function( rc )
+    local g, U, lc; 
+    g := Representative( rc ); 
+    U := ActingDomain( rc ); 
+    lc := LeftCoset( g^-1, U ); 
+    SetAssociatedRightCoset( lc, rc ); 
+    return lc;
+end); 
+
+InstallMethod( PrintString, "for a left coset", true, [ IsLeftCosetObj ], 0,
+function( d )
+    return STRINGIFY( "LeftCoset(\<",
+                      PrintString( Representative(d) ), ",\>",
+                      PrintString( ActingDomain(d) ), ")" );
+end);
+
+InstallMethod( PrintObj, "for a left coset", true, [ IsLeftCosetObj ], 0,
+function( d )
+    Print( PrintString( d ) );
+end);
+
+InstallMethod( ViewString, "for a left coset", true, [ IsLeftCosetObj ], 0,
+function( d )
+    return STRINGIFY( "LeftCoset(\<",
+                      ViewString( Representative(d) ), ",\>",
+                      ViewString( ActingDomain(d) ), ")" );
+end);
+
+InstallMethod( ViewObj, "for a left coset", true, [ IsLeftCosetObj ], 0,
+function( d )
+    Print( ViewString( d ) );
+end);
+
+InstallMethod( \=, "for two left cosets", IsIdenticalObj,
+    [ IsLeftCoset, IsLeftCoset ], 0,
+function( lc1, lc2 ) 
+    return AssociatedRightCoset( lc1 ) = AssociatedRightCoset( lc2 ); 
+end);
+
+InstallMethod( \in, "element and LeftCoset", true,
+    [ IsMultiplicativeElementWithInverse, IsLeftCoset ], 100,
+function( g, lc )
+    Print( "Hey, hey!  The method for \in in lcset.gi is being called!\n" ); 
+    return g^-1 in AssociatedRightCoset( lc );
+end);
+
+InstallOtherMethod( \*, "element and LeftCoset", true,
+    [ IsMultiplicativeElementWithInverse, IsLeftCoset ], 0,
+function( g, lc )
+    return LeftCoset( g * Representative( lc ), ActingDomain( lc ) );
+end);
+
+InstallMethod( Size, "for a left coset", true, [ IsLeftCoset ], 0,
+function( lc ) 
+    return Size( AssociatedRightCoset( lc ) ); 
+end);
+
+InstallOtherMethod( IsBiCoset, "for a left coset", true, [ IsLeftCoset ], 0,
+function( lc ) 
+    return IsBiCoset( AssociatedRightCoset( lc ) ); 
+end);
+
+InstallOtherMethod( AsSet, "for a left coset", true, [ IsLeftCoset ], 0,
+function( lc ) 
+    local L; 
+    L := AsSet( AssociatedRightCoset( lc ) ); 
+    return Set( L, g -> g^-1 ); 
+end);
+
+InstallOtherMethod( Intersection2, "for two left cosets", true, 
+    [ IsLeftCoset, IsLeftCoset ], 0,
+function( lc1, lc2 ) 
+    local L; 
+    L := Intersection2( AssociatedRightCoset( lc1 ),
+                        AssociatedRightCoset( lc2 ) ); 
+    return Set( L, g -> g^-1 ); 
```

Comment:
I think we recently change `Intersection` of right cosets to return a right coset again, so perhaps `return Inverse(L)` instead (assuming that `Inverse` is changed, as I suggested, to return the "associated" left coset)

Reply:
Yes, that works well in gapdev, but not in 4r11.  Best to leave as is for now?